lib/web/apiserver.go

```diff
@@ -2328,3 +2336,35 @@ func makeTeleportClientConfig(ctx *SessionContext) (*client.Config, error) {
 
 	return config, nil
 }
+
+type ssoRequestParams struct {
+	clientRedirectURL string
+	connectorID       string
+	csrfToken         string
+}
+
+func parseSSORequestParams(r *http.Request) (*ssoRequestParams, error) {
+	query := r.URL.Query()
+
+	clientRedirectURL := query.Get("redirect_url")
+	if clientRedirectURL == "" {
+		return nil, trace.BadParameter("Missing redirect_url query parameter")
+	}
+
+	connectorID := query.Get("connector_id")
+	if connectorID == "" {
+		return nil, trace.BadParameter("Missing connector_id query parameter.")
```

Comment:
Do not capitalize and remove the dot. Same below.

Reply:
revised